client/my-sites/checkout/upsell-nudge/professional-email-upsell/index.tsx

```diff
@@ -17,30 +17,43 @@ import {
 	transformMailboxForCart,
 	validateMailboxes,
 } from 'calypso/lib/titan/new-mailbox';
+import type { MinimalRequestCartProduct } from '@automattic/shopping-cart';
+import type { TranslateResult } from 'i18n-calypso';
+import type { ChangeEvent } from 'react';
 
 import './style.scss';
 
-const noopWithCallback = ( ignored, callback = () => {} ) => {
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noopWithCallback = ( cartItem: MinimalRequestCartProduct, callback = () => {} ) => {
 	callback();
 };
 
-const ProfessionalEmailFeature = ( { children } ) => {
+const ProfessionalEmailFeature = ( { children }: { children: TranslateResult } ) => {
 	return (
 		<li>
-			<Gridicon icon="checkmark" size="18" />
+			<Gridicon icon="checkmark" size={ 18 } />
 			<span>{ children }</span>
 		</li>
 	);
 };
 
+type ProfessionalEmailUpsellProps = {
+	currencyCode: string;
+	domainName: string;
+	handleClickAccept: ( action: string ) => void;
+	handleClickDecline: () => void;
+	productCost?: number | null;
+	setCartItem: ( cartItem: MinimalRequestCartProduct, callback: () => void ) => void;
+};
+
 const ProfessionalEmailUpsell = ( {
 	currencyCode,
 	domainName,
 	handleClickAccept,
 	handleClickDecline,
 	productCost,
 	setCartItem = noopWithCallback,
```

Comment:
I wonder why we define a default value for that prop since we always provide `setCartItem` when calling that component.